lib/install/server/server.go

```diff
@@ -116,6 +134,12 @@ type Executor interface {
 	Complete(operationKey ops.SiteOperationKey) error
 }
 
+// DebugReporter allows to capture the operation state
+type DebugReporter interface {
```

Comment:
Made this into a separate interface so it can be optionally implemented (and it is only implemented by the installer but not agents).